File: pasos.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    int num;
    printf("Introduce un número entero positivo: ");
    scanf("%d", &num);

    if (num <= 0) {
        printf("Número inválido. Introduce un entero positivo.\n");
        return 1;
    }

    printf("Secuencia para %d:\n", num);
    int steps = 0;
    while (num != 1) {
        printf("%d -> ", num);
        if (num % 2 == 0) {
            num = num / 2;
        } else {
            num = 3 * num + 1;
        }
        steps++;
    }
    printf("1\n");
    printf("Finalizada en %d pasos.\n", steps);

    return 0;
}

File: pasos.py
# Solicita al usuario un número entero positivo
n = int(input("Introduce un número entero positivo: "))

# Verifica si el número es positivo
if n <= 0:
    print("Número inválido. Introduce un entero positivo.")
else:
    # Imprime el número inicial
    print(f"Secuencia para {n}:")
    steps = 0
    while n != 1:
        print(f"{n} -> ", end="")
        if n % 2 == 0:
            n = n // 2
        else:
            n = 3 * n + 1
        steps += 1
    print("1")
    print(f"Finalizada en {steps} pasos.")

